README.md

```diff
@@ -250,7 +242,7 @@ class Button extends Component {
 ```
 
 ---
-If we want to use composed `theme` in many lifecycle hooks or, for instance, in methods that can be called dozens of times in short period of time, like in react-motion, we can manually check for changing `theme` props and compose final theme in `getDerivedStateFromProps`. To avoid even looking theme up in cache in hot functions.
+If we want to use composed `theme` in many lifecycle hooks or, for instance, in methods that can be called dozens of times quickly, like in react-motion, we perform a check when `theme` prop change and compose a theme state declaration in `getDerivedStateFromProps` e.g.  "this.state.theme" to avoid using a cached theme in hot functions.
```

Comment:
Maybe we ca clarify *when **incoming** `theme` prop change and compose a **result** theme state declaration*

Reply:
Not sure I get what is `theme state declaration`, though

Reply:
Also, `"this.state.theme" to avoid using a cached theme`.
In both cases we might use the same cached theme, whether it's from getDSFP or from manual getThemeFromProps. The point was to avoid searching in cache in hot functions and save it in state instead

Reply:
// maybe this

If we want to use composed `theme` in many lifecycle hooks or, for instance, in methods that can be called dozens of times quickly,
like in react-motion, we can manually check for changing `theme` props and compose a state theme in `getDerivedStateFromProps`.
By having a state theme, searching the cache in hot functions can be avoided.

Reply:
I like it!